hazelcast/src/test/java/com/hazelcast/cache/CacheBasicAbstractTest.java

```diff
@@ -274,175 +272,144 @@ public void testIterator() {
 
     @Test
     public void testIteratorRemove() {
-        ICache cache = createCache();
+        ICache<Integer, Integer> cache = createCache();
         int size = 1111;
         for (int i = 0; i < size; i++) {
             cache.put(i, i);
         }
 
-        Iterator<Cache.Entry<Integer, Integer>> iter = cache.iterator();
-        while (iter.hasNext()) {
-            iter.next();
-            iter.remove();
+        Iterator<Cache.Entry<Integer, Integer>> iterator = cache.iterator();
+        while (iterator.hasNext()) {
+            iterator.next();
+            iterator.remove();
         }
         assertEquals(0, cache.size());
     }
 
     @Test(expected = IllegalStateException.class)
     public void testIteratorIllegalRemove() {
-        ICache cache = createCache();
+        ICache<Integer, Integer> cache = createCache();
         int size = 10;
         for (int i = 0; i < size; i++) {
             cache.put(i, i);
         }
 
-        Iterator<Cache.Entry<Integer, Integer>> iter = cache.iterator();
-        if (iter.hasNext()) {
-            iter.remove();
+        Iterator<Cache.Entry<Integer, Integer>> iterator = cache.iterator();
+        if (iterator.hasNext()) {
+            iterator.remove();
         }
     }
 
     @Test
-    public void testIteratorDuringInsertion() throws InterruptedException {
-        final AtomicBoolean stop = new AtomicBoolean(false);
-        final ICache cache = createCache();
-        int size = 1111;
+    @SuppressWarnings("WhileLoopReplaceableByForEach")
+    public void testIteratorDuringInsertion_withoutEviction() {
+        final ICache<Integer, Integer> cache = createCache();
+        final int size = 1111;
         for (int i = 0; i < size; i++) {
             cache.put(i, i);
         }
 
-        final Thread thread = new Thread() {
-            public void run() {
-                Random rand = new Random();
-                while (!stop.get()) {
-                    int i = rand.nextInt();
-                    try {
-                        cache.put(i, i);
-                        LockSupport.parkNanos(1);
-                    } catch (Throwable ignored) {
-                    }
-                }
+        AbstractCacheWorker worker = new AbstractCacheWorker() {
+            @Override
+            void doRun(Random random) {
+                int i = random.nextInt(size);
```

Comment:
Default max-size policy is `ENTRY_COUNT` and its default size is `10.000`. Based on our estimation based max-size evaluation algorithm, partition's max size, which is used internally, can be calculated via 
`EntryCountCacheMaxSizeChecker::calculateMaxPartitionSize(int maxEntryCount, int partitionCount)`. Eviction is done for a partition if its size is reached to max partition size.

So, in theory, it is guaranteed that if the element count is less than max partition size, there is no eviction even though data is distributed as unbalanced. But if the element count is bigger than max partition size, there may be eviction due to unbalanced data (data can be stored only one partition).

Also, in theory, it is guaranteed that if the element count is bigger than `max_partition_size * partition_count`,
there is eviction even though data is distributed as perfectly balanced.

As a result, if we want to be sure that there is no eviction, we should set the `size` to max partition size, so it is guaranteed that there is no eviction. To find the max partition size, the following code snippet can be used:
``` java
final int partitionCount = ...;
final int maxPartitionSize = 
	EntryCountCacheMaxSizeChecker.calculateMaxPartitionSize(
		EvictionConfig.DEFAULT_MAX_ENTRY_COUNT, partitionCount);
		
```

For example, max partition size and max global size with default size (which is `10.000`) and default partition count (which is `271`):
```
Max partition size : 55
Max global size    : 14905
```

I know, in this test, we are giving keys sequencially from `0` to `size` and there is no unbalanced state, it would be nice if we don't trust this fact in the test and set the size to max partition size. 

Also, for the `testIteratorDuringInsertion_withoutEviction`, I prefer putting half (or less than size) of the elements before worker thread start and so worker thread can make an insertion (not update).